packages/nextjs/tsup.config.ts

```diff
@@ -55,6 +60,31 @@ export default defineConfig(overrideOptions => {
     sourcemap: false,
   };
 
+  /**
+   * We vendor certain dependencies to control the output and minimize transitive dependency surface area.
+   */
+  const vendorsEsm: Options = {
+    ...esm,
+    bundle: true,
+    minify: true,
+    entry: ['./src/vendor/**'],
```

Comment:
Create entry points only for JS files
```suggestion
    entry: ['./src/vendor/*.js'],
```